test/apiv2/23-containersArchive.at

```diff
@@ -0,0 +1,28 @@
+# -*- sh -*-
+#
+# test more container-related endpoints
+#
+
+podman pull $IMAGE &>/dev/null
+
+# Ensure clean slate
+podman rm -a -f &>/dev/null
+
+CTR="ArchiveTestingCtr"
+
+TMPD=$(mktemp -d)
+pushd "${TMPD}"
+echo "Hello!" > "hello.txt" &> /dev/null
+tar --format=posix -cvf "hello.tar" "hello.txt" &> /dev/null
+popd
+
+HELLO_TAR="${TMPD}/hello.tar"
+
+podman run -d --name "${CTR}" "${IMAGE}" top
+
+t HEAD "containers/nonExistentCtr/archive?path=%2F" 404
+t HEAD "containers/${CTR}/archive?path=%2Fnon%2Fexistent%2Fpath" 404
+t HEAD "containers/${CTR}/archive?path=%2Fetc%2Fpasswd" 200
+
+curl "http://127.0.0.1:$PORT/containers/${CTR}/archive?path=%2Ftmp" -X PUT --upload-file "${HELLO_TAR}"
```

Comment:
Seems there is no support for `PUT` yet. Also the testing suite expects that bodies are `JSON`s but in this case its tar stream. So I am just doing really simple test here.

Another thing is that it seems I can't use `podman-exec`. I tried:
```bash
if ! podman exec "${CTR}" cat "/etc/passwd" ; then
  echo FAIL;
fi
```
And it prints "FAIL" but it shouldn't the passwd should be there.

Reply:
Also how can I easily access headers?

Reply:
Sorry for not checking before. I think that would require some bash wizardry. I suggest for now to limit tests to `HEAD` only. We can tackle the rest once the PR is merged.

Reply:
I added some more tests, but I am kinda bypassing test utility. Have you ever considered using `Go` rather than `bash` for `api` testing?